src/main/scala/uk/gov/nationalarchives/consignmentexport/Bagit.scala

```diff
@@ -0,0 +1,31 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.nio.file.Path
+import java.util
+import java.util.UUID
+
+import cats.effect.IO
+import gov.loc.repository.bagit.creator.BagCreator
+import gov.loc.repository.bagit.domain.Bag
+import gov.loc.repository.bagit.hash.{StandardSupportedAlgorithms, SupportedAlgorithm}
+import gov.loc.repository.bagit.verify.BagVerifier
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import uk.gov.nationalarchives.consignmentexport.Utils._
+
+import scala.jdk.CollectionConverters._
+import scala.language.postfixOps
+
+class Bagit(bagInPlace: (Path, util.Collection[SupportedAlgorithm], Boolean) => Bag,
+           isComplete: (Bag, Boolean) => Unit)(implicit val logger: SelfAwareStructuredLogger[IO]) {
+
+  def createBag(consignmentId: UUID, rootLocation: String, includeHiddenFiles: Boolean): IO[Unit] = for {
+    bag <- IO.pure(bagInPlace(s"$rootLocation/$consignmentId".toPath, List(StandardSupportedAlgorithms.SHA256: SupportedAlgorithm).asJavaCollection, includeHiddenFiles))
+    _ <- IO.pure(isComplete(bag, includeHiddenFiles))
+    _ <- logger.info("Bagit export complete")
```

Comment:
I've added the consignment id here and any other log lines where it's available. I haven't logged any of the file ids because if we get a consignment with a lot of files, we'll get a lot of log spam which will be hard to parse through. So long as the consignment id is there we should be able to recreate any problems.